File: maven-root/mtsa/src/main/java/MTSAEnactment/ar/uba/dc/lafhis/enactment/robot/N6/robotProtocol/CommunicationStartMessage.java
package MTSAEnactment.ar.uba.dc.lafhis.enactment.robot.N6.robotProtocol;

public class CommunicationStartMessage extends Message {	
	public static int MESSAGE_ID				= 1;
	
	public CommunicationStartMessage(int to, int from){
		super(MESSAGE_ID, 1, to, from);
	}
	
	public CommunicationStartMessage(String serializedMessage){
		super(serializedMessage);
	}
	
	@Override
	public int getMessageLength() {
		return super.getMessageLength();
	}
	
	
	@Override
	protected void readData(String inputString){
	}

	@Override
	protected String writeData(){
		return "";
	}

	@Override
	protected boolean hasReply() {
		return true;
	}
	
}

File: maven-root/mtsa/src/main/java/RobotProtocol/src/main/java/robotProtocol/messages/CommunicationStartMessage.cpp
/*
 * CommunicationStartMessage.cpp
 *
 *  Created on: Jan 31, 2013
 *      Author: mariano
 */

#include "CommunicationStartMessage.h"

namespace RobotProtocol {

CommunicationStartMessage CommunicationStartMessage::staticMessage = CommunicationStartMessage(0,0);

Message *CommunicationStartMessage::deserialize(char *serializedMessage){
	CommunicationStartMessage::staticMessage.initialize(serializedMessage);
	return &(CommunicationStartMessage::staticMessage);
}

CommunicationStartMessage::CommunicationStartMessage(int to, int from)
:Message(MESSAGE_ID, 1, to, from){}


CommunicationStartMessage::CommunicationStartMessage(char *serializedMessage){
	initialize(serializedMessage);
}

CommunicationStartMessage::~CommunicationStartMessage() {
}

int CommunicationStartMessage::getMessageLength(){
	return Message::getMessageLength();
}

void CommunicationStartMessage::readData(char **serializedMessageIndex){}
void CommunicationStartMessage::writeData(char **serializedMessageIndex){}

} /* namespace RobotProtocol */
